Feature Read Time

justingreerbbi opened the pull request:
Merge contains cleanup of naming convention, commenting, spacing and logic operators.. YODA!!!

justingreerbbi committed:
initial realtime hook

justingreerbbi committed:
Cleanup for calculation for read time

justingreerbbi committed:
Add hh_get_post_read_time function to display read time in front-end

Not tested yet

justingreerbbi committed:
Updated function name

Ready for testing

justingreerbbi committed:
Renamed and moved plugin to directory

justingreerbbi committed:
Cleanup of mess

justingreerbbi committed:
Removing template test of get_the_read_time()

hlashbrooke commented on plugins/helphub-read-time.php:
```diff
@@ -0,0 +1,85 @@
+
+	// No post revisions
+	if ( wp_is_post_revision( $post_id ) )
+		return;
```
Coding standards: Please use curly braces for all clauses, even if they're only one line.

hlashbrooke commented on plugins/helphub-read-time.php:
```diff
@@ -0,0 +1,85 @@
+
+	// If the post type is not found then return
+	if ( ! in_array( $post->post_type, $calculate_for_posts ) )
+		return;
```
Coding standards: Please use curly braces for all clauses, even if they're only one line.

hlashbrooke commented on plugins/helphub-read-time.php:
```diff
@@ -0,0 +1,85 @@
+	// Converts read time seconds into minutes
+	$converted_readtime = 0 == $read_time ? '~1' : round( $read_time / 60 );
+
+	return sprintf( esc_html__( 'Read Time: %s Min', 'helphub' ), $converted_readtime );
```
I would say 'Reading time' is a more natural way of saying this, but I'm open to opinions on that.

Apina replied:
Can I also recommend using Minute or Minutes rather than Min - purely for English as a second language users, so the meaning is absolutely clear.

hlashbrooke replied:
Good point. @justingreerbbi I imagine that `_n()` would come in handy here.

justingreerbbi committed:
added curly brackets and changed Read Time to Reading Time

justingreerbbi committed:
added _n() for reading time minutes

get_the_read_time() now parses read time and returns full singular or
plural format. Its just goober english this way :)

justingreerbbi committed:
space fix

jonathanbossenger commented on plugins/helphub-read-time.php:
```diff
@@ -0,0 +1,87 @@
+defined( 'ABSPATH' ) or die( 'No script kiddies please!' );
+
+// Adds the read time hook when saving a post
+add_action( 'save_post', 'hh_calulate_and_update_post_read_time', 10, 3 );
```
Being super pedantic but this should be hh_calculate_and_update_post_read_time

justingreerbbi replied:
Good Catch. Yes it should be!

jonathanbossenger commented on plugins/helphub-read-time.php:
```diff
@@ -0,0 +1,87 @@
+ * @param  boolean 	$update  Is this an update or new.
+ * @return void
+ */
+function hh_calulate_and_update_post_read_time( $post_id, $post, $update ) {
```
See previous comment

justingreerbbi committed:
Spelling fix

updated function name and action callback

justingreerbbi committed:
Minor Restructuring

Created hh_get_readtime for private access and broken up hh_ public
functions for native feel of echo and return.

justingreerbbi committed:
logic update

Updated hh_get_readtime to check if is_null and to ensure the $post_id
arg is numeric.

Default still is global $post->ID

justingreerbbi committed:
fixed confusion

justingreerbbi committed:
tab fix

justingreerbbi committed:
tab fix

justingreerbbi committed:
tab fix

justingreerbbi committed:
updated examples

hlashbrooke commented on plugins/helphub-read-time.php:
```diff
@@ -0,0 +1,115 @@
+	$hh_reading_time = hh_get_readtime( $post_id );
+
+	// Filter the time before it is converted
+	$read_time = apply_filters( 'hh_post_read_time_' . $post_id , $hh_reading_time );
```
I would say change this so that the filter name is constant and the post ID comes in as the second parameter:

```
$read_time = apply_filters( 'hh_post_read_time', $hh_reading_time, $post_id );
```

That would just make it easier to work with if you wanted to (for example) filter all the reading time values instead of only filtering it on a single post.

justingreerbbi committed:
updated hh_post_read_time filter logic
